linkcheck/HtmlParser/htmlsax.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf8 -*-
+# Copyright (C) 2000-2018 Petr Dlouhy
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Replacement for the built-in Html parser
+"""
+
+from bs4 import BeautifulSoup
+
+from ..containers import ListDict
+
+
+class Parser(object):
+    html_doc = u""
+    handler = None
+    encoding = "iso8859-1"
+
+    def __init__(self, handler):
+        self.handler = handler
+
+    def feed(self, feed_text):
+        if type(feed_text) == bytes:
+            feed_text = feed_text.decode(self.encoding)
+        self.html_doc += feed_text
+
+    def reset(self):
+        self.html_doc = u""
+
+    def flush(self):
+        soup = BeautifulSoup(self.html_doc, 'html.parser')
```

Comment:
Oh, how about this instead: make self.html_doc a StringIO(), write to it in feed(), and here do html_doc = self.html_doc.getvalue(), and then do the if isinstance(html_doc, bytes): html_doc = html_doc.decode(self.encoding)?

Also, what if the document encoding is not actually ISO-8859-1?